### PR TITLE
chore: Bump pgbouncer from 1.17.0 to 1.18.0

### LINKED ISSUE #1195: Update to PgBouncer 1.18
PgBouncer 1.18 has been released.
The full changelog is available here: https://www.pgbouncer.org/changelog.html#pgbouncer-118x

The new container image has been automatically generated and can be found at `ghcr.io/cloudnative-pg/pgbouncer:1.18.0`.
We should change cnpg to use v1.18 by default.

### PR DESCRIPTION
**Pull request by PascalBourdier**
Closes #1195

### CODE REVIEW

**Review by mirakladp — APPROVED** [review]

File: docs/src/release_notes/old/v1.15.md [hunk 1]
```diff
@@ -211,7 +211,7 @@ Features:
 - Support for the `AWS_SESSION_TOKEN` authentication token in AWS S3 through
   the `sessionToken` option
 - Default image name for PgBouncer in `Pooler` pods set to
-  `quay.io/enterprisedb/pgbouncer:1.17.0`
```
Comment (sxd) [inline docs/src/release_notes/old/v1.15.md:214]: Please do not change the old documentation, it's been kept like that for historical reasons =)

**Review by sxd — CHANGES_REQUESTED** [review]
Please revert the change in the old release notes

**Review by sxd — APPROVED** [review]

**Comment by sxd** [comment]
All the test failed here due to changes in the observability https://github.com/cloudnative-pg/cloudnative-pg/pull/1201

**Comment by PascalBourdier** [comment]
> All the test failed here due to changes in the observability #1201

@sxd I have updated metric names according this change: https://github.com/pgbouncer/pgbouncer/pull/717

**Comment by sxd** [comment]
@PascalBourdier let's wait for the test then! :D

**Comment by sxd** [comment]
@PascalBourdier something else is failing you can check the tests here https://github.com/cloudnative-pg/cloudnative-pg/actions/runs/3712641545

**Comment by PascalBourdier** [comment]
> @PascalBourdier something else is failing you can check the tests here https://github.com/cloudnative-pg/cloudnative-pg/actions/runs/3712641545

Yes, I see it.
But `cl_active_cancel_req` and `cl_waiting_cancel_req` metrics seem presents:
```
pgbouncer=# SHOW POOLS;
-[ RECORD 1 ]---------+----------
database              | pgbouncer
user                  | pgbouncer
cl_active             | 1
cl_waiting            | 0
cl_active_cancel_req  | 0
cl_waiting_cancel_req | 0
sv_active             | 0
sv_active_cancel      | 0
sv_being_canceled     | 0
sv_idle               | 0
sv_used               | 0
sv_tested             | 0
sv_login              | 0
maxwait               | 0
maxwait_us            | 0
pool_mode             | statement
```

I don't understand why they are not found:
```
  Expected
      <[]string | len:12, cap:20>: [
          "cnpg_pgbouncer_collection_duration_seconds{collector=\"Collect.up\"} 0.00617826",
          "cnpg_pgbouncer_collections_total 1",
          "cnpg_pgbouncer_last_collection_error 0",
          "cnpg_pgbouncer_lists_dns_pending 0",
          "cnpg_pgbouncer_lists_dns_queries 0",
          "cnpg_pgbouncer_lists_free_clients 49",
          "cnpg_pgbouncer_lists_pools 1",
          "cnpg_pgbouncer_lists_used_servers 0",
          "cnpg_pgbouncer_lists_users 2",
          "cnpg_pgbouncer_stats_avg_query_time{database=\"pgbouncer\"} 0",
          "cnpg_pgbouncer_stats_avg_recv{database=\"pgbouncer\"} 0",
          "cnpg_pgbouncer_stats_total_query_count{database=\"pgbouncer\"} 3",
      ]
  to have length 14
```
`cnpg_pgbouncer_pools_pool_mode` seems miss too ...

**Comment by PascalBourdier** [comment]
Ok, I think https://github.com/cloudnative-pg/pgbouncer-containers should be updated first !

**Review by sxd — CHANGES_REQUESTED** [review]
Tests still failing

**Review by phisco — APPROVED** [review]

**Review by sxd — APPROVED** [review]

**Pull request merged**